### PR TITLE
Entrust source plugin

### PR DESCRIPTION
**Pull request by sirferl**
This implements the entrust source plugin.
It gets all active Entrusts certificates

### CODE REVIEW

File: lemur/plugins/lemur_entrust/plugin.py [hunk 1]
```diff
@@ -313,10 +313,88 @@ class EntrustSourcePlugin(SourcePlugin):
...
+            )
+            try:
+                data = json.loads(response.content)
+            except ValueError:
+                # catch an empty jason object here
+                data = {'response': 'No detailed message'}
+            status_code = response.status_code
+            if status_code > 399:
+                raise Exception(f"ENTRUST error: {status_code}\n{data['errors']}")
+            # current_app.logger.info(f"recevied: {data['summary']}")
```
Comment (hosseinsh) [inline lemur/plugins/lemur_entrust/plugin.py:376]: shall we drop this line?

File: lemur/plugins/lemur_entrust/plugin.py [hunk 2]
```diff
@@ -313,10 +313,88 @@ class EntrustSourcePlugin(SourcePlugin):
...
+            for c in data["certificates"]:
+                download_url = "{0}{1}".format(
+                    host, c["uri"]
+                )
+                cert_response = self.session.get(download_url)
+                certificate = json.loads(cert_response.content)
+                # current_app.logger.info(f"Result: {certificate}")
```
Comment (hosseinsh) [inline lemur/plugins/lemur_entrust/plugin.py:383]: same here

File: lemur/plugins/lemur_entrust/plugin.py [hunk 3]
```diff
@@ -313,10 +313,88 @@ class EntrustSourcePlugin(SourcePlugin):
...
+                    "body": certificate["endEntityCert"],
+                    "serial": serial,
+                    "external_id": str(certificate["trackingId"]),
+                }
+                certs.append(cert)
+            if data["summary"]["limit"] * offset >= data["summary"]["total"]:
+                break
+            else:
+                offset += 1
+        current_app.logger.info(f"Result: {certs}")
```
Comment (hosseinsh) [inline lemur/plugins/lemur_entrust/plugin.py:396]: wondering if logging all the certs is a good idea, or rather the total number of discovered certs?

File: lemur/plugins/lemur_entrust/plugin.py [hunk 4]
```diff
@@ -313,10 +313,88 @@ class EntrustSourcePlugin(SourcePlugin):
 
     author = "sirferl"
     author_url = "https://github.com/sirferl/lemur"
+    options = [
```
Comment (hosseinsh) [inline lemur/plugins/lemur_entrust/plugin.py:316]: is this needed, and if so, shall we make a better use of it than dummy?

**Review by hosseinsh — COMMENTED** [review]
thanks for the quick PR @sirferl

File: lemur/plugins/lemur_entrust/plugin.py (hunk 4 above)
Comment (sirferl) [reply lemur/plugins/lemur_entrust/plugin.py:316]: Hi @hosseinsh ,
this relates to https://github.com/Netflix/lemur/issues/2105 . Someday I will fix this problem as well...
So sadly , we have to leave it in at the moment.

File: lemur/plugins/lemur_entrust/plugin.py (hunk 1 above)
Comment (sirferl) [reply lemur/plugins/lemur_entrust/plugin.py:376]: I'll takes this out.

File: lemur/plugins/lemur_entrust/plugin.py (hunk 2 above)
Comment (sirferl) [reply lemur/plugins/lemur_entrust/plugin.py:383]: I_ll takes this out.

File: lemur/plugins/lemur_entrust/plugin.py (hunk 3 above)
Comment (sirferl) [reply lemur/plugins/lemur_entrust/plugin.py:396]: I'll take this out

File: lemur/plugins/lemur_entrust/plugin.py [hunk 5]
```diff
@@ -318,10 +319,91 @@ class EntrustSourcePlugin(SourcePlugin):
...
+                cert_response = self.session.get(download_url)
+                certificate = json.loads(cert_response.content)
+                # normalize serial
+                serial = str(int(certificate["serialNumber"], 16))
+                cert = {
+                    "body": certificate["endEntityCert"],
+                    "serial": serial,
+                    "external_id": str(certificate["trackingId"]),
+                    "csr": certificate["csr"],
+                    "owner": certificate["tracking"]["requesterEmail"],
```
Comment (hosseinsh) [inline lemur/plugins/lemur_entrust/plugin.py:396]: excellent idea to track owner and add description

**Review by hosseinsh — APPROVED** [review]

**Comment by hosseinsh** [comment]
Thanks again @sirferl for the PR. 
We can improve the plugin in the future and also add testing, time permitting. 
merging shortly. 

PS: I have also added a minor language change directly on the branch, hope okay with you.

**Comment by sirferl** [comment]
your text is fine with me! Thanks for approving.

**Pull request merged**